File: tests/test-stream.py
# Copyright (c) 2018, Red Hat Inc.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at:
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

import sys

import ovs.stream
import ovs.util


def main(argv):
    if len(argv) < 2:
        ovs.util.ovs_fatal(0,
                           "usage: %s REMOTE [SSL_KEY] [SSL_CERT] [SSL_CA]",
                           argv[0],
                           )
    remote = argv[1]

    if remote.startswith("ssl:"):
        if len(argv) < 5:
            ovs.util.ovs_fatal(
                0,
                "usage with ssl: %s REMOTE [SSL_KEY] [SSL_CERT] [SSL_CA]",
                argv[0],
            )
        ovs.stream.SSLStream.ssl_set_ca_cert_file(argv[4])
        ovs.stream.SSLStream.ssl_set_certificate_file(argv[3])
        ovs.stream.SSLStream.ssl_set_private_key_file(argv[2])

    err, stream = ovs.stream.Stream.open_block(
            ovs.stream.Stream.open(remote), 10000)

    if err or stream is None:
        sys.exit(1)

    sys.exit(0)


if __name__ == '__main__':
    main(sys.argv)

File: tests/test-stream.c
/*
 * Copyright (c) 2018 Ilya Maximets <i.maximets@samsung.com>
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at:
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

#include <config.h>

#include "fatal-signal.h"
#include "openvswitch/vlog.h"
#include "stream.h"
#include "util.h"

VLOG_DEFINE_THIS_MODULE(test_stream);

int
main(int argc, char *argv[])
{
    int error;
    struct stream *stream;

    fatal_ignore_sigpipe();
    set_program_name(argv[0]);

    if (argc < 2) {
        ovs_fatal(0, "usage: %s REMOTE", argv[0]);
    }

    error = stream_open_block(stream_open(argv[1], &stream, DSCP_DEFAULT),
                              10000, &stream);
    if (error) {
        VLOG_ERR("stream_open_block(%s) failure: %s",
                 argv[1], ovs_strerror(error));
    }
    stream_close(stream);
    return (error || !stream) ? 1 : 0;
}
